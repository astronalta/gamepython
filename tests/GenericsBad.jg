
Generic[:a] < Object {
}

Alpha < Object {
}

Beta < Object {
}

Generic2[:a,:b] < Object {
    attr1 Generic[:a] = Generic[:b]() # Not allowed
    attr2 Generic[:a] = Generic[String]() # Not allowed
    attr3 Generic[String] = Generic[:a]() # Not allowed
    fun() {
        a Generic[:a] = Generic[:b]() # Not allowed
        b Generic[:a] = Generic[String]() # Not allowed
        c Generic[String] = Generic[:a]() # Not allowed
    }
}


main() Int {
    #g = Generic[Alpha]()
    #g = Generic[Beta]()
    ret 0
}
