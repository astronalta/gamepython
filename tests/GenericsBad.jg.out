GenericsBad.jg:12:5: Expression does not conform to type 'Generic[:a]'
GenericsBad.jg:13:5: Expression does not conform to type 'Generic[:a]'
GenericsBad.jg:14:5: Expression does not conform to type 'Generic[String]'
GenericsBad.jg:16:25: Expression does not conform to type 'Generic[:a]'
GenericsBad.jg:17:25: Expression does not conform to type 'Generic[:a]'
GenericsBad.jg:18:29: Expression does not conform to type 'Generic[String]'
